martinMutuma/def-politico
# #164725852 UI auth form validation

### What does this PR do?
This PR validates the UI auth fields so that a user can know beforehand of the mistakes they made before sending the information to the server.

### Description of the Tasks to be accomplished

- [x] - Create validation functions
- [x] - Validate login functionality
- [x] - Validate signup functionality

### How should this be manually tested?
1. Clone this repo and checkout to `ch/164725852-auth-ui-validation` branch
2. Go to login or signup and try creating an account or logging in with a poorly formatted email
3. In the login page, try leaving either the email field or the password field blank, you will instantly see a message on the submit button telling you to make sure that the fields are entered before logging in.


### Any background context you want to add
This will reduce the time it takes for a user to correct the information they are filling in order to authenticate and as a result, make the app feel more responsive and fast.

### Screenshots
![image](https://user-images.githubusercontent.com/12128153/54622186-33fa0680-4a7a-11e9-934f-b55dbd19d9c0.png)

![image](https://user-images.githubusercontent.com/12128153/54622218-3f4d3200-4a7a-11e9-855b-218319a9bea9.png)

![image](https://user-images.githubusercontent.com/12128153/54622242-4b38f400-4a7a-11e9-9079-e9df87f76e79.png)
